calvinpete/Glokeeper-backend
# #162725255 User should be able to signup, login and reset password while receiving emails

#### What does this PR do?
This PR sets up the authentication capabilities of the application.

#### Description of Task to be completed?
- Set up signup, login, request password reset, reset password APIs
- Construct Django email templates to send to emails to the users
- Back the APIs with serializers and models

#### How should this be manually tested?
- First things first, fetch this branch by running the command `git fetch ft-user-signup-162725255` in the terminal.
- Check into this branch by running `git checkout ft-user-signup-162725255`
- Activate your virtual environment by running `source venv/bin/activate`
- Delete the current database by running `drop database DBName;` in the psql shell
- Create a new database by running `create database DBName;` in the psql shell
- Update the .env file with the database name
- Migrate by running `python manage.py migrate`
- Start the application by running `python manage.py runserver`
- To test the endpoints , open postman and run the postman collection below

#### What are the relevant pivotal tracker stories?
[#162725255](https://www.pivotaltracker.com/story/show/162725255)

####Postman Collection
[![Run in Postman](https://run.pstmn.io/button.svg)](https://god.postman.co/run-collection/bd9c58dc8d87920d1e3f?action=collection%2Fimport)
